src/sink.rs

```diff
@@ -0,0 +1,101 @@
+    format,
+    frame::{self, Video},
+    media::Type,
+    Dictionary,
+};
+
+use crate::state::Pipeline;
+
+#[allow(dead_code)]
+pub struct RtpSink {
+    port: u16,
+}
+
+impl RtpSink {
+    pub fn new(pipeline: Arc<Pipeline>, port: u16) -> Self {
+        let port_clone = port;
+        thread::spawn(move || {
+            RtpSink::start(pipeline, port_clone).unwrap();
+        });
+        Self { port }
+    }
+
+    fn start(pipeline: Arc<Pipeline>, port: u16) -> Result<()> {
+        let sdp_filepath = PathBuf::from(format!("/tmp/sdp_input_{}.sdp", port));
+        let mut file = File::create(&sdp_filepath)?;
+        file.write_all(
+            format!(
+                "\
+                    v=0\n\
+                    o=- 0 0 IN IP4 127.0.0.1\n\
+                    s=No Name\n\
+                    c=IN IP4 127.0.0.1\n\
+                    m=video {} RTP/AVP 96\n\
+                    a=rtpmap:96 H264/90000\n\
+                    a=fmtp:96 packetization-mode=1\n\
+                    a=rtcp-mux\n\
+                ",
+                port
+            )
+            .as_bytes(),
+        )?;
+        let mut input_ctx = format::input_with_dictionary(
+            &sdp_filepath,
+            Dictionary::from_iter([("protocol_whitelist", "file,udp,rtp")]),
+        )?;
+        let input = input_ctx
+            .streams()
+            .best(Type::Video)
+            .ok_or(ffmpeg_next::Error::StreamNotFound)?;
+        let input_index = input.index();
+        let decoder = Context::from_parameters(input.parameters())?;
+        let decoder = decoder.decoder();
+        let h264_codec = codec::decoder::find(Id::H264).unwrap();
+        let mut decoder = decoder.open_as(h264_codec)?;
+
+        let mut decoded_frame = frame::Video::empty();
+        for (stream, packet) in input_ctx.packets() {
+            if stream.index() != input_index {
+                eprintln!("Received frame from unknown stream, skipping");
+                continue;
+            }
+            decoder.send_packet(&packet)?;
+            while decoder.receive_frame(&mut decoded_frame).is_ok() {
+                let frame = match frame_from_av(&mut decoded_frame) {
+                    Ok(frame) => frame,
+                    Err(err) => {
+                        eprintln!("Dropping frame: {err}");
+                        continue;
+                    }
+                };
+                pipeline.push_input_data(port.into(), frame);
+            }
+        }
+
+        Ok(())
+    }
+}
+
+fn frame_from_av(decoded: &mut Video) -> Result<Frame> {
+    if decoded.format() != format::pixel::Pixel::YUV420P {
+        return Err(anyhow!("only YUV420P is supported"));
+    }
+    Ok(Frame {
+        data: YuvData {
+            y_plane: bytes::Bytes::copy_from_slice(decoded.data(0)),
+            u_plane: bytes::Bytes::copy_from_slice(decoded.data(1)),
+            v_plane: bytes::Bytes::copy_from_slice(decoded.data(2)),
+        },
```

Comment:
data returns &[u8]. there is no way to take ownership of that data. To avoid this copy I would have to pass ffpmeg frame all the way to the renderer code, which would complicate api.

Note that I'm only copying the data, but memory for that frame is allocated only once, because ffmpeg frame is reused. Addi tonally this approach would allow us to change that in the future to have pool of allocated buffers. I didn't do any benchmarks, but I believe allocation is more expensive than copy.

Reply:
Ok. It's not crucial at this point. We can leave it here as it is. 

Alternatively, in future, we can use a buffers pool or `renderer` that could have function like:
```rust
pub struct Planes {
    y_plane: &[u8],
    u_plane: &[u8],
    v_plane: &[u8]
}

pub fn wgpu_texture_from_slices(&self, planes: &Planes, resolution: Resolution) -> wgpu::Texture {...}
``` 
and we can keep `wgpu::Texture` later on (but we will be using a lot of VRAM with this solution).
If we create GPU decoding/encoding, then it'll replace it anyway.